range-v3 should be /W4-clean on MSVC

When I compile a simple program like this:

	#include <range/v3/algorithm.hpp>
	#include <vector>

	int main()
	{
		std::vector<int> v{3, 2, 5, 8, 6, 6, 1, 6, 2};
		ranges::sort( v );
	}

 with VS2017 15.9 with W4, I get a lot of warnings along the lines of 

>[...] range\v3\algorithm\sort.hpp(92): warning C4459: declaration of 'next' hides global declaration
  [...] range/v3/utility/iterator.hpp(265): note: see declaration of 'ranges::v3::next'

While this is certainly not a bug, it would be nice, if the ranges-v3 headers would compile warning free on a "reasonable warning level" (and yes, I consider W4 to be reasonable, but of course you might disagree).

The variable `data` is inside an inline namespace, so I'm unsure about this...
Is this a MSVC bug (and I should report a bug there) or a can I submit a merge request here?

    1> \range-v3\include\range\v3\detail\variant.hpp(706,1): warning C4459:  declaration of 'data' hides global declaration
    1> \range-v3\include\range\v3\detail\variant.hpp(706,1): warning C4459:         {
    1> \range-v3\include\range\v3\detail\variant.hpp(706,1): warning C4459: ^
    1> \range-v3\include\range\v3\data.hpp(84,13): message :  see declaration of 'ranges::v3::CPOs::data'
    1> \range-v3\include\range\v3\data.hpp(84,13): message :             RANGES_INLINE_VARIABLE(data_detail::data_fn, data) 
    1> \range-v3\include\range\v3\data.hpp(84,13): message :             ^

--- a/include/range/v3/algorithm/sort.hpp
+++ b/include/range/v3/algorithm/sort.hpp
@@ -100,12 +100,12 @@ namespace ranges
         inline void unguarded_linear_insert(I last, iter_value_t<I> val, C & pred,
                                             P & proj)
         {
-            I next = prev(last);
-            while(invoke(pred, invoke(proj, val), invoke(proj, *next)))
+            I next_ = prev(last);
+            while(invoke(pred, invoke(proj, val), invoke(proj, *next_)))
             {
-                *last = iter_move(next);
-                last = next;
-                --next;
+                *last = iter_move(next_);
+                last = next_;
+                --next_;
             }
             *last = std::move(val);
         }

